libs/parallelism/execution/include/hpx/execution/algorithms/detail/single_result.hpp

```diff
@@ -6,10 +6,15 @@
 
 #pragma once
 
+#include <hpx/config.hpp>
 #include <hpx/type_support/pack.hpp>
 
 #include <type_traits>
 
+#if defined(HPX_HAVE_CXX17_STD_VARIANT)
+#include <variant>
+#endif
```

Comment:
Probably unrelated to this PR, but should we add serialization support for `std::variant` (could be simply taken from here: https://github.com/STEllAR-GROUP/phylanx/blob/master/phylanx/util/serialization/variant.hpp)?

Reply:
If you think it would be useful, sure.

Reply:
I'll create a separate PR, then. So far there was no need for this as we didn't use the type ourselves.

Reply:
NWM, we already have that :/